src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs

```diff
@@ -19412,6 +19412,142 @@ public void AbstractInSubmission()
+}";
+            var forwardingCompilation = CreateCompilationWithCustomILSource(string.Empty, forwardingIL);
+
+            forwardingCompilation.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(16484, "https://github.com/dotnet/roslyn/issues/16484")]
+        public void MultipleForwardsToDifferentAssembliesWhileReferencingThemShouldErrorOut()
+        {
+            var forwardingCode = @"
+namespace ForwardingNamespace
+{
+    public class Program
+    {
+        public static void Main()
+        {
+            new Destination.TestClass();
+        }
+    }
+}";
+            var forwardingIL = @"
+.assembly extern Destination1 { }
+.assembly extern Destination2 { }
+.class extern forwarder Destination.TestClass
+{
+	.assembly extern Destination1
+}
+.class extern forwarder Destination.TestClass
+{
+	.assembly extern Destination2
+}";
+            var forwardingCompilation = CreateCompilationWithCustomILSource(forwardingCode, forwardingIL);
+
+            forwardingCompilation.VerifyDiagnostics(
+                // (8,29): error CS1577: Type 'Destination.TestClass' is forwarded to multiple assemblies: 'Destination1' and 'Destination2'
+                //             new Destination.TestClass();
+                Diagnostic(ErrorCode.ERR_TypeForwardedToMultipleAssemblies, "TestClass").WithArguments("Destination.TestClass", "Destination1", "Destination2").WithLocation(8, 29));
+        }
+
+        [Fact, WorkItem(16484, "https://github.com/dotnet/roslyn/issues/16484")]
+        public void MultipleForwardsThatResultInTheSameAssemblyDoNotErrorOut()
+        {
+            var forwardedToCode1 = @"
+namespace Destination
+{
+    public class TestClass {}
+}";
+            var forwardedToReference1 = CreateCompilationWithMscorlib(forwardedToCode1, assemblyName: "Destination1").EmitToImageReference();
+
+            var forwardedToCode2 = @"[assembly: System.Runtime.CompilerServices.TypeForwardedTo(typeof(Destination.TestClass))]";
+            var forwardedToCompilation2 = CreateCompilationWithMscorlib(forwardedToCode2, new MetadataReference[] { forwardedToReference1 }, assemblyName: "Destination2");
+            var forwardedToReference2 = forwardedToCompilation2.EmitToImageReference();
+
+            var forwardingCode = @"
+namespace ForwardingNamespace
+{
+    public class Program
+    {
+        public static void Main()
+        {
+            new Destination.TestClass();
+        }
+    }
+}";
+            var forwardingIL = @"
+.assembly extern Destination1 { }
+.assembly extern Destination2 { }
+.class extern forwarder Destination.TestClass
+{
+	.assembly extern Destination1
+}
+.class extern forwarder Destination.TestClass
+{
+	.assembly extern Destination2
+}";
+            var forwardingCompilation = CreateCompilationWithCustomILSource(forwardingCode, forwardingIL, new MetadataReference[] { forwardedToReference1, forwardedToReference2 });
+
+            forwardingCompilation.VerifyDiagnostics();
+        }
     }
```

Comment:
Note to self: In Binder.NotFound(), there are two paths that the check can go. With or without type qualifiers. Add another test to ensure that both paths are tested.